### DISCUSSION TITLE
Use decimal format for numeric constants

### DISCUSSION THREAD

**Discussion by jonasraoni** [pkp/pkp-lib#7985] [Proposals]
**Describe the problem you would like to solve**
Searching for numeric values in the code is a bit complex sometimes, given that constants might be expressed in a hexadecimal format, and everywhere else (logs, database, XML, etc.) they are written in decimal.

**Describe the solution you'd like**
Replace the hexadecimal constants by decimal values or leave a comment with an alternative format.

From:
```php
public const ROLE_ID_MANAGER = 0x00000010;
```

To (as they are probably part of a binary set, maybe the binary format provides a better visualization):

```php
public const ROLE_ID_MANAGER = 0b10000; //16
public const ROLE_ID_MANAGER = 16; //0b10000
```

**Who is asking for this feature?**
Developer

**Comment by NateWr** [comment]
:+1: it'd be nice to change this in the code base. In the meantime, there's a little tool available to help: https://github.com/pkp/pkp-lib/blob/main/tools/constants.php

Get all defined constants and their values:

```
php lib/pkp/tools/constants.php
```

Get the name of all constants defined with a specific value:

```
php lib/pkp/tools/constants.php 16
```
